src/Symfony/Bundle/SecurityBundle/Command/UserPasswordEncoderCommand.php

```diff
@@ -86,13 +89,18 @@ protected function execute(InputInterface $input, OutputInterface $output)
         }
 
         if (!$salt) {
-            $saltQuestion = $this->createSaltQuestion($input, $output);
-            $salt = $helper->ask($input, $output, $saltQuestion);
+            if ($input->isInteractive()) {
+                $saltQuestion = $this->createSaltQuestion($input, $output);
+                $salt = $helper->ask($input, $output, $saltQuestion);
+            } else {
+                $salt = $this->generateSalt($output);
+            }
+
         }
 
         $output->writeln("\n <comment>Encoders are configured by user type in the security.yml file.</comment>");
 
-        if (!$userClass) {
+        if (!$userClass && $input->isInteractive()) {
             $userClassQuestion = $this->createUserClassQuestion($input, $output);
```

Comment:
@ogizanagi is it really a useless question ? Each user class can use a separate encoder.